lib/pbench/cli/agent/commands/tools/list.py

```diff
@@ -32,39 +51,69 @@ def execute(self):
         else:
             groups = self.groups
 
+        with_option = False
         if not self.context.name:
             host_tools = {}
             for group in groups:
                 host_tools[group] = {}
-                for path in self.gen_tools_group_dir(group).glob("*/**"):
-                    if self.context.with_option:
-                        host_tools[group][path.name] = [
-                            p.read_text() for p in self.tools(path)
-                        ]
-                    else:
-                        host_tools[group][path.name] = [p for p in self.tools(path)]
+                try:
+                    for path in self.gen_tools_group_dir(group).glob("*/**"):
+                        host = path.name
+                        if self.context.with_option:
+                            with_option = True
+                            host_tools[group][host] = [
+                                (p, self.chomp((path / p).read_text())) for p in self.tools(path)
+                            ]
+                        else:
+                            host_tools[group][host] = [p for p in self.tools(path)]
+                except BadToolGroup:
+                    self.logger.error("Tool group does not exist: %s", group)
+                    return 1
             if host_tools:
-                for k, v in host_tools.items():
-                    for h, t in v.items():
-                        print("%s: %s %s" % (k, h, t))
+                self.print_results(host_tools, with_option)
         else:
             # List the groups which include this tool
             group_list = []
+            tool = self.context.name
+            options = {}
             for group in groups:
-                tg_dir = self.gen_tools_group_dir(group)
+                try:
+                    tg_dir = self.gen_tools_group_dir(group)
+                except BadToolGroup:
+                    self.logger.error("Tool group does not exist: %s", group)
```

Comment:
`BaseCommand.gen_tools_group_dir()` (`ToolGroup.verify_tool_group()`) uses `strict=True` path resolution, so why do we need the `tg_dir.exists()` check below?

Reply:
It was there in the original code and I didn't touch it, but practically as well as theoretically, it's not needed: I haven't seen this message pop up during my testing.

Reply:
Fixed.